### ISSUE TITLE
unable to open config file from search list /var/lib/rook/rook-ceph/rook-ceph.config

### ISSUE THREAD

**Issue by fhchina** [rook/rook#2551]
**Is this a bug report or feature request?**
*Bug Report

**Deviation from expected behavior:**
pvc pending state. error event: persistentvolume-controller  waiting for a volume to be created, either by external provisioner "ceph.rook.io/block" or manually created by system administrator

**Expected behavior:**
pvc get created, not pending.

**How to reproduce it (minimal and precise):**

**Environment**:
* OS (e.g. from /etc/os-release):
NAME="Ubuntu"
VERSION="18.04.1 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.1 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic

* Kernel (e.g. `uname -a`):
Linux ubuntu 4.15.0-43-generic #46-Ubuntu SMP Thu Dec 6 14:45:28 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

* Cloud provider or hardware configuration:
OpenStack

* Rook version (use `rook version` inside of a Rook Pod):
rook: v0.9.0-79.g1a1ffdd

* Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.2", GitCommit:"cff46ab41ff0bb44d8584413b598ad8360ec1def", GitTreeState:"clean", BuildDate:"2019-01-13T23:16:58Z", GoVersion:"go1.11.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.2", GitCommit:"cff46ab41ff0bb44d8584413b598ad8360ec1def", GitTreeState:"clean", BuildDate:"2019-01-10T23:28:14Z", GoVersion:"go1.11.4", Compiler:"gc", Platform:"linux/amd64"}

* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
On Premise

* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/Rook/master/toolbox.html)):
HEALTH_OK

**Comment by fhchina** [comment]
kubectl describe pvc jenkins
Name:          jenkins
Namespace:     jx
StorageClass:  rook-ceph
Status:        Pending
Volume:
Labels:        app=jenkins
               chart=jenkins-0.10.32
               heritage=Tiller
               release=jenkins-x
Annotations:   control-plane.alpha.kubernetes.io/leader:
                 {"holderIdentity":"bfedb076-1efe-11e9-b697-56e96830e2e1","leaseDurationSeconds":15,"acquireTime":"2019-01-24T09:15:00Z","renewTime":"2019-...
               volume.beta.kubernetes.io/storage-provisioner: ceph.rook.io/block
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:
Access Modes:
VolumeMode:    Filesystem
Events:
  Type     Reason              Age                 From                                                                                        Message
  ----     ------              ----                ----                                                                                        -------
  Normal   Provisioning        41m (x14 over 56m)  ceph.rook.io/block rook-ceph-operator-86554fd8d-msc7v bfedb076-1efe-11e9-b697-56e96830e2e1  External provisioner is provisioning volume for claim "jx/jenkins"
  Warning  ProvisioningFailed  41m (x14 over 56m)  ceph.rook.io/block rook-ceph-operator-86554fd8d-msc7v bfedb076-1efe-11e9-b697-56e96830e2e1  Failed to provision volume with StorageClass "rook-ceph": Failed to create rook block image replicapool/pvc-86c92f3f-1fb8-11e9-802e-fa163ece3c89: failed to create image pvc-86c92f3f-1fb8-11e9-802e-fa163ece3c89 in pool replicapool of size 32212254720: Failed to complete '': exit status 1. global_init: unable to open config file from search list /var/lib/rook/rook-ceph/rook-ceph.config
. output:
  Normal     ExternalProvisioning  26m (x330 over 56m)   persistentvolume-controller  waiting for a volume to be created, either by external provisioner "ceph.rook.io/block" or manually created by system administrator
  Normal     ExternalProvisioning  3m10s (x82 over 23m)  persistentvolume-controller  waiting for a volume to be created, either by external provisioner "ceph.rook.io/block" or manually created by system administrator
Mounted By:  jenkins-7b69dd8964-ftkpb

**Comment by fhchina** [comment]
kubectl describe storageclasses.storage.k8s.io rook-ceph
Name:            rook-ceph
IsDefaultClass:  Yes
Annotations:     kubectl.kubernetes.io/last-applied-configuration={"apiVersion":"storage.k8s.io/v1","kind":"StorageClass","metadata":{"annotations":{"storageclass.kubernetes.io/is-default-class":"true"},"name":"rook-ceph"},"parameters":{"blockPool":"replicapool","clusterNamespace":"rook-ceph","fstype":"xfs"},"provisioner":"ceph.rook.io/block"}
,storageclass.kubernetes.io/is-default-class=true
Provisioner:           ceph.rook.io/block
Parameters:            blockPool=replicapool,clusterNamespace=rook-ceph,fstype=xfs
AllowVolumeExpansion:  <unset>
MountOptions:          <none>
ReclaimPolicy:         Delete
VolumeBindingMode:     Immediate
Events:                <none>

**Comment by fhchina** [comment]
The rook/ceph cluster works at first. But after I delete all of the block images using Ceph Dashboard. The pvc binding is always pending.

**Comment by fhchina** [comment]
almost all the operation in rook-operator's log will print: 
Failed to complete '': exit status 1. global_init: unable to open config file from search list /var/lib/rook/rook-ceph/rook-ceph.config

I use the following command to view log:
kubectl -n rook-ceph-system logs `kubectl -n rook-ceph-system -l app=rook-operator get pods -o jsonpath='{.items[*].metadata.name}'`

**Comment by fhchina** [comment]
2019-01-23 11:05:07.679681 I | rookcmd: starting Rook v0.9.1 with arguments '/usr/local/bin/rook ceph operator'
2019-01-23 11:05:07.680220 I | rookcmd: flag values: --alsologtostderr=false, --help=false, --log-level=INFO, --log_backtrace_at=:0, --log_dir=, --logtostderr=true, --mon-healthcheck-interval=45s, --mon-out-timeout=5m0s, --stderrthreshold=2, --v=0, --vmodule=
2019-01-23 11:05:07.764261 I | cephcmd: starting operator
2019-01-23 11:05:08.900804 I | op-agent: getting flexvolume dir path from FLEXVOLUME_DIR_PATH env var
2019-01-23 11:05:08.900835 I | op-agent: flexvolume dir path env var FLEXVOLUME_DIR_PATH is not provided. Defaulting to: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/
2019-01-23 11:05:08.900855 I | op-agent: discovered flexvolume dir path from source default. value: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/
2019-01-23 11:05:08.900880 I | op-agent: no agent mount security mode given, defaulting to '%s' modeAny
2019-01-23 11:05:10.084723 I | op-agent: rook-ceph-agent daemonset already exists, updating ...
2019-01-23 11:05:10.195302 I | op-discover: rook-discover daemonset already exists, updating ...
2019-01-23 11:05:13.066675 I | operator: rook-provisioner ceph.rook.io/block started using ceph.rook.io flex vendor dir
I0123 11:05:13.067024      10 controller.go:407] Starting provisioner controller bfedb076-1efe-11e9-b697-56e96830e2e1!
2019-01-23 11:05:13.071849 I | operator: rook-provisioner rook.io/block started using rook.io flex vendor dir
2019-01-23 11:05:13.071886 I | op-cluster: start watching clusters in all namespaces
I0123 11:05:13.072749      10 controller.go:407] Starting provisioner controller c1a2289f-1efe-11e9-b697-56e96830e2e1!
2019-01-23 11:05:13.266022 I | op-cluster: skipping watching for legacy rook cluster events (legacy cluster CRD probably doesn't exist): the server could not find the requested resource (get clusters.ceph.rook.io)
2019-01-23 11:05:13.465555 I | op-cluster: starting cluster in namespace rook-ceph
2019-01-23 11:05:13.603436 I | op-k8sutil: verified the ownerref can be set on resources
2019-01-23 11:05:13.668396 I | op-k8sutil: waiting for job rook-ceph-detect-version to complete...
2019-01-23 11:07:18.662536 E | op-cluster: unknown ceph major version. failed to complete version job. failed to detect job . the server was unable to return a response in the time allotted, but may still be processing the request (get jobs.batch rook-ceph-detect-version)
I0124 01:48:14.168780      10 controller.go:1072] scheduleOperation[delete-pvc-78dfa871-1ef6-11e9-802e-fa163ece3c89[8c2c6e5d-1ef6-11e9-802e-fa163ece3c89]]
2019-01-24 01:48:14.683101 I | op-provisioner: Deleting volume pvc-78dfa871-1ef6-11e9-802e-fa163ece3c89
2019-01-24 01:48:14.683539 I | exec: Running command: rbd rm replicapool/pvc-78dfa871-1ef6-11e9-802e-fa163ece3c89 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring
I0124 01:48:15.110030      10 controller.go:1072] scheduleOperation[delete-pvc-78cfa62b-1ef6-11e9-802e-fa163ece3c89[8c1d502b-1ef6-11e9-802e-fa163ece3c89]]
2019-01-24 01:48:15.163178 I | op-provisioner: Deleting volume pvc-78cfa62b-1ef6-11e9-802e-fa163ece3c89
2019-01-24 01:48:15.163487 I | exec: Running command: rbd rm replicapool/pvc-78cfa62b-1ef6-11e9-802e-fa163ece3c89 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring
E0124 01:48:15.164846      10 controller.go:1043] Deletion of volume "pvc-78dfa871-1ef6-11e9-802e-fa163ece3c89" failed: Failed to delete rook block image replicapool/pvc-78dfa871-1ef6-11e9-802e-fa163ece3c89: failed to delete image pvc-78dfa871-1ef6-11e9-802e-fa163ece3c89 in pool replicapool: Failed to complete '': exit status 1. global_init: unable to open config file from search list /var/lib/rook/rook-ceph/rook-ceph.config
. output:
E0124 01:48:15.165155      10 goroutinemap.go:150] Operation for "delete-pvc-78dfa871-1ef6-11e9-802e-fa163ece3c89[8c2c6e5d-1ef6-11e9-802e-fa163ece3c89]" failed. No retries permitted until 2019-01-24 01:48:15.665062414 +0000 UTC m=+52989.143042559 (durationBeforeRetry 500ms). Error: "Failed to delete rook block image replicapool/pvc-78dfa871-1ef6-11e9-802e-fa163ece3c89: failed to delete image pvc-78dfa871-1ef6-11e9-802e-fa163ece3c89 in pool replicapool: Failed to complete '': exit status 1. global_init: unable to open config file from search list /var/lib/rook/rook-ceph/rook-ceph.config\n. output: "
E0124 01:48:15.467575      10 controller.go:1043] Deletion of volume "pvc-78cfa62b-1ef6-11e9-802e-fa163ece3c89" failed: Failed to delete rook block image replicapool/pvc-78cfa62b-1ef6-11e9-802e-fa163ece3c89: failed to delete image pvc-78cfa62b-1ef6-11e9-802e-fa163ece3c89 in pool replicapool: Failed to complete '': exit status 1. global_init: unable to open config file from search list /var/lib/rook/rook-ceph/rook-ceph.config
. output:
E0124 01:48:15.468498      10 goroutinemap.go:150] Operation for "delete-pvc-78cfa62b-1ef6-11e9-802e-fa163ece3c89[8c1d502b-1ef6-11e9-802e-fa163ece3c89]" failed. No retries permitted until 2019-01-24 01:48:15.967831529 +0000 UTC m=+52989.445811683 (durationBeforeRetry 500ms). Error: "Failed to delete rook block image replicapool/pvc-78cfa62b-1ef6-11e9-802e-fa163ece3c89: failed to delete image pvc-78cfa62b-1ef6-11e9-802e-fa163ece3c89 in pool replicapool: Failed to complete '': exit status 1. global_init: unable to open config file from search list /var/lib/rook/rook-ceph/rook-ceph.config\n. output: "
I0124 01:48:25.666397      10 controller.go:1072] scheduleOperation[delete-pvc-78cfa62b-1ef6-11e9-802e-fa163ece3c89[8c1d502b-1ef6-11e9-802e-fa163ece3c89]]
I0124 01:48:25.666541      10 controller.go:1072] scheduleOperation[delete-pvc-78dfa871-1ef6-11e9-802e-fa163ece3c89[8c2c6e5d-1ef6-11e9-802e-fa163ece3c89]]
2019-01-24 01:48:26.447181 I | op-provisioner: Deleting volume pvc-78cfa62b-1ef6-11e9-802e-fa163ece3c89
2019-01-24 01:48:26.447244 I | exec: Running command: rbd rm replicapool/pvc-78cfa62b-1ef6-11e9-802e-fa163ece3c89 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/roo:

**Comment by Stolr** [comment]
Hi there ,

Same issue on my side from a fresh install , my pvc are staying in pending state.

```
 Warning  ProvisioningFailed    14m (x2 over 14m)   ceph.rook.io/block_rook-ceph-operator-54476dc55f-5phxf_b94ef833-7b9a-11e9-9e1f-4ec24f97590d  failed to provision volume with StorageClass "rook-ceph-block": Failed to create rook block image replicapool/pvc-2f31b4c8-7b17-11e9-9157-001dd8f11c10: failed to create image pvc-2f31b4c8-7b17-11e9-9157-001dd8f11c10 in pool replicapool of size 8589934592: Failed to complete '': exit status 1. global_init: unable to open config file from search list /var/lib/rook/rook-ceph/rook-ceph.config


```

**Comment by abheyogy** [comment]
Same issue here, having a kops installed cluster on AWS.

**Comment by Amos-85** [comment]
I deal with the same issue with on-premise environment.
rook version 0.9

**Comment by chrono2002** [comment]
same here

**Comment by pkavajin** [comment]
Can reproduce this issue when having multiple cephclusters, might be related?

**Comment by travisn** [comment]
If this is failing when there are multiple clusters, this should be fixed in v1.0.2. See the release notes for https://github.com/rook/rook/releases/tag/v1.0.2 and #3261. 
@fhchina @abheyogy @Amos-85 @chrono2002 Can you confirm if you had multiple clusters or if you are still seeing the issue?

**Comment by pkavajin** [comment]
Actually, I'm already running v1.0.2.
Feels a bit fishy, seems like the operator can provision volumes either for one cluster or another, either I'm getting the error for the one cluster or the other, with a restart sometimes switching the cluster it handles. (Can still be total unrelated tho).

Is there some debug information I could provide which might help to give some clearness whats up here?

**Comment by leseb** [comment]
> If this is failing when there are multiple clusters, this should be fixed in v1.0.2. See the release notes for [v1.0.2](https://github.com/rook/rook/releases/tag/v1.0.2) and #3261.
> @fhchina @abheyogy @Amos-85 @chrono2002 Can you confirm if you had multiple clusters or if you are still seeing the issue?

It'll be helpful to get a reply to that question. Thanks.
